twisted/python/_setup.py

```diff
@@ -135,133 +157,96 @@ def __init__(self, *args, **kwargs):
 
 
 
-def setup(**kw):
-    """
-    An alternative to distutils' setup() which is specially designed
-    for Twisted subprojects.
-
-    @param conditionalExtensions: Extensions to optionally build.
-    @type conditionalExtensions: C{list} of L{ConditionalExtension}
-    """
-    return _setup(**get_setup_args(**kw))
+# The C extensions used for Twisted.
+_EXTENSIONS = [
+    ConditionalExtension(
+        "twisted.test.raiser",
+        ["twisted/test/raiser.c"],
+        condition=lambda _: _isCPython),
 
+    ConditionalExtension(
+        "twisted.internet.iocpreactor.iocpsupport",
+        ["twisted/internet/iocpreactor/iocpsupport/iocpsupport.c",
+         "twisted/internet/iocpreactor/iocpsupport/winsock_pointers.c"],
+        libraries=["ws2_32"],
+        condition=lambda _: _isCPython and sys.platform == "win32"),
 
-def get_setup_args(**kw):
-    if 'cmdclass' not in kw:
-        kw['cmdclass'] = {'build_scripts': build_scripts_twisted}
+    ConditionalExtension(
+        "twisted.python._sendmsg",
+        sources=["twisted/python/_sendmsg.c"],
+        condition=lambda _: not _PY3 and sys.platform != "win32"),
 
-    if "conditionalExtensions" in kw:
-        extensions = kw["conditionalExtensions"]
-        del kw["conditionalExtensions"]
-
-        if 'ext_modules' not in kw:
-            # This is a workaround for distutils behavior; ext_modules isn't
-            # actually used by our custom builder.  distutils deep-down checks
-            # to see if there are any ext_modules defined before invoking
-            # the build_ext command.  We need to trigger build_ext regardless
-            # because it is the thing that does the conditional checks to see
-            # if it should build any extensions.  The reason we have to delay
-            # the conditional checks until then is that the compiler objects
-            # are not yet set up when this code is executed.
-            kw["ext_modules"] = extensions
+    ConditionalExtension(
+        "twisted.runner.portmap",
+        ["twisted/runner/portmap.c"],
+        condition=lambda builder: not _PY3 and
+                                  builder._check_header("rpc/rpc.h")),
+    ]
 
-        class my_build_ext(build_ext_twisted):
-            conditionalExtensions = extensions
-        kw.setdefault('cmdclass', {})['build_ext'] = my_build_ext
-    return kw
 
 
-def getVersion(base):
+def get_setup_args():
```

Comment:
I would like this to be `def getSetupArgs(extensions=_EXTENSIONS):` to fit the coding standard, and mean you can pass `_EXTENSIONS` in explicitly without monkeypatching.